mpc_local_planner/include/mpc_local_planner/mpc_local_planner_ros.h

```diff
@@ -186,9 +186,17 @@ class MpcLocalPlannerROS : public nav_core::BaseLocalPlanner, public mbf_costmap
     /**
      * @brief Get the current robot footprint/contour model
      * @param nh const reference to the local ros::NodeHandle
+     * @param costmap_ros reference to an intialized instance of costmap_2d::Costmap2dROS
      * @return Robot footprint model used for optimization
      */
-    static RobotFootprintModelPtr getRobotFootprintFromParamServer(const ros::NodeHandle& nh);
+    static RobotFootprintModelPtr getRobotFootprintFromParamServer(const ros::NodeHandle& nh, costmap_2d::Costmap2DROS* costmap_ros = nullptr);
+
+    /**
+     * @brief Get the current robot footprint/contour model
+     * @param costmap_ros reference to an intialized instance of costmap_2d::Costmap2dROS
+     * @return Robot footprint model used for optimization
+     */
+    static RobotFootprintModelPtr getRobotFootprintFromCostmap2d(costmap_2d::Costmap2DROS* costmap_ros = nullptr);
```

Comment:
I know it's a little fussy, but here I would like to pass the costmap as const reference, as nobody would invoke this method with a nullptr as there is no use-case for it.
```suggestion
    static RobotFootprintModelPtr getRobotFootprintFromCostmap2d(const costmap_2d::Costmap2DROS& costmap_ros);
```

Reply:
oh okay, i guess you didn't mark it as const because `getRobotFootprintPolygon()` in costmap_2d is not const either. Good, we keep it this way. Thank you for this PR and the changes! I'll merge this now.